net/xia/ppal_ether/main.c

```diff
@@ -297,13 +361,18 @@ static int main_newroute(struct xip_ppal_ctx *ctx, struct fib_xid_table *xtbl,
 	fxid_init(xtbl, &mether->xem_common, id, XRTABLE_MAIN_INDEX, 0);
 	xdst_init_anchor(&mether->xem_anchor);
 	INIT_LIST_HEAD(&mether->interface_common_addr);
+	seqlock_init(&mether->cached_hdr.hh_lock);
 
 	mether->host_interface = out_interface;
 	memmove(mether->ha, cfg->xfc_lladdr, cfg->xfc_lladdr_len);
 	attach_neigh_to_interface(mether);
 
-	WARN_ON(ether_rt_iops->fxid_add_locked(&bucket, xtbl,
-					       &mether->xem_common));
+	mether->cached_hdr.hh_len = 0;
+	WARN_ON(mfxid_hh_init(mether));
```

Comment:
Once `mfxid_hh_init()` is replace with `xia_ether_header_cache()`, the line `mether->cached_hdr.hh_len = 0;` is no longer needed.